src/SFML/Window/EglContext.cpp

```diff
@@ -287,67 +297,100 @@ EGLConfig EglContext::getBestConfig(EGLDisplay display, unsigned int bitsPerPixe
 {
     EglContextImpl::ensureInit();
 
-    // Set our video settings constraint
-    const EGLint attributes[] =
-        {EGL_BUFFER_SIZE,
-         static_cast<EGLint>(bitsPerPixel),
-         EGL_DEPTH_SIZE,
-         static_cast<EGLint>(settings.depthBits),
-         EGL_STENCIL_SIZE,
-         static_cast<EGLint>(settings.stencilBits),
-         EGL_SAMPLE_BUFFERS,
-         settings.antialiasingLevel ? 1 : 0,
-         EGL_SAMPLES,
-         static_cast<EGLint>(settings.antialiasingLevel),
-         EGL_SURFACE_TYPE,
-         EGL_WINDOW_BIT | EGL_PBUFFER_BIT,
-         EGL_RENDERABLE_TYPE,
-         EGL_OPENGL_ES_BIT,
-         EGL_NONE};
-
-    EGLint    configCount;
-    EGLConfig configs[1];
-
-    // Ask EGL for the best config matching our video settings
-    eglCheck(eglChooseConfig(display, attributes, configs, 1, &configCount));
-
-    // TODO: This should check EGL_CONFORMANT and pick the first conformant configuration.
-
-    return configs[0];
+    // Determine the number of available configs
+    EGLint configCount;
+    eglCheck(eglGetConfigs(display, nullptr, 0, &configCount));
+
+    // Retrieve the list of available configs
+    auto configs = std::make_unique<EGLConfig[]>(static_cast<std::size_t>(configCount));
+
+    eglCheck(eglGetConfigs(display, configs.get(), configCount, &configCount));
+
+    // Evaluate all the returned configs, and pick the best one
+    int       bestScore = 0x7FFFFFFF;
+    EGLConfig bestConfig{};
+
+    for (std::size_t i = 0; i < static_cast<std::size_t>(configCount); ++i)
```

Comment:
Do we know how big this number may get? Could it ever get big enough to cause problems?

Reply:
It's not that big... typically in the hundreds on a desktop system with a powerful GPU, in a mobile context probably less.